src/core/ColorUtil.h

```diff
@@ -24,8 +25,16 @@ void rgbtohsv(float r, float g, float b, float& h, float& s, float& v);
 
 std::optional<Color4f> parse_hex_color(const std::string& hex);
 
+std::optional<Color4f> parse_web_color(const std::string& col);
+
+std::optional<Color4f> parse_color(const std::string& col);
+
+Color4f getColor(const std::string& col, const Color4f& defaultcolor = Color4f(0.0f, 0.0f, 0.0f));
```

Comment:
getColor() is never called without providing a default, so I'd skip the default value.
However, since the only function of this extra parameter is to convert from an optional<Color4f> to a Color4f, why not simply expose the optional? That way clients are responsible for dealing with it, as well as providing a contextually correct error message.

..which means you can simply kill getColor() and call parse_color() everywhere; one less function to deal with.

Reply:
I am planning to use `getColor` for PDF and SVG exports and there I will be using the default. Here's an example from a closed pull request: https://github.com/openscad/openscad/pull/5845/files#r2059363559 (The function is called `parse_color` in that version.)

Reply:
Ah, that makes more sense, as a common place to emit the warning. I'd still kill the default value though, as black isn't necessarily a universal color fallback.